### PR TITLE
Lambda-like support for callbacks

### PR DESCRIPTION
**Pull request by KavyanshKhaitan2**
Added rich callback support for callbacks. This should still keep most of the uses still working except the ones with spaces in them.

Also adding more documentation for the above feature.

Issues resolved: #32

### CODE REVIEW

**Comment by KavyanshKhaitan2** [comment]
Showcase video:
https://github.com/ObaraEmmanuel/Formation/assets/73186427/fa918c10-8cd1-4f77-ac3a-2404c49a5340

**Comment by KavyanshKhaitan2** [comment]
Huh. the "workflow tests" failed because I used `f"{varName=}"` syntax for testing inside `if __name__ == "__main__"`... So pushed that on!

**Comment by ObaraEmmanuel** [comment]
Now I remember why I avoided type hinting. It is always changing and not consistent. Your code only works for 3.10 because of the type hinting.

**Comment by ObaraEmmanuel** [comment]
I would recommend you drop all type hints for now

**Comment by KavyanshKhaitan2** [comment]
> I would recommend you drop all type hints for now

see; I wont drop ALL type hinting; I would like to keep some basic type hinting for sure.

Again; pushing a commit...

File: docs/callback.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+**⚠ This article is a stub ⚠**
+
+Previously, Callbacks could not take any arguments in formation-studio. With this being implemented, formation-studio can take limited arguments.
+
+As of right now, callbacks can be used with Events and ``command`` fields.
+
+Format
+=======
+**Format**: ``{func_name} {arg1}, {arg2}, kw1=1, kw2=2``
+
```
Comment (ObaraEmmanuel) [inline docs/callback.rst:13]: Remove this file. I will incorporate proper documentation later.

File: formation/callback_parse.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    return command_func, args, kwargs
+
+# if __name__ == "__main__":
+#     command = "btn 1, 2, 3, 'hello this is a world!', kw1='my kwarg'"
+#     output = parse(command)
+#     command_func, command_args, command_kwargs = output
+#     print(f"{command_kwargs = }")
+#     print(f"{command_args = }")
+#     print(f"{command_func = }")
+
```
Comment (ObaraEmmanuel) [inline formation/callback_parse.py:50]: Remove this module as well. I think the parsing is simple enough now to fit in `utils.py`.

File: formation/loader.py [hunk 3]
```diff
@@ -425,24 +428,30 @@ def on_keypress(self, event):
...
+                    partial_handler = partial(handler, *parsed[1], **parsed[2]) # parsed[1] is function args/ parsed[2] is function kwargs.
                     widget.bind(
                         event.get("sequence"),
-                        handler,
+                        partial_handler,
                         event.get("add")
                     )
                 else:
-                    logger.warning("Callback '%s' not found", event.get("handler"))
+                    logger.warning("Callback '%s' not found in event_map", parsed[0])
```
Comment (ObaraEmmanuel) [inline formation/loader.py:443]: Avoid changing the wording of any logged messages.

File: formation/loader.py [hunk 4]
```diff
@@ -425,24 +428,30 @@ def on_keypress(self, event):
...
+            handler = callback_map.get(parsed[0]) # parsed[0] is the function name.
             if handle_method is None:
                 raise ValueError("Handle method is None, unable to apply binding")
             if handler is not None:
-                handle_method(**{prop: handler})
+                partial_handler = partial(handler, *parsed[1], **parsed[2]) # parsed[1] is function args/ parsed[2] is function kwargs.
+                handle_method(**{prop: partial_handler})
             else:
-                logger.warning("Callback '%s' not found", val)
+                logger.warning("Callback '%s' not found in command_map", val)
```
Comment (ObaraEmmanuel) [inline formation/loader.py:454]: Avoid changing the wording of any logged messages.

**Review by ObaraEmmanuel — CHANGES_REQUESTED** [review]

File: formation/callback_parse.py (hunk 2 above)
Comment (KavyanshKhaitan2) [reply formation/callback_parse.py:50]: i think you are right...

**Review by ObaraEmmanuel — APPROVED** [review]
I have polished up the code. It still does not pass some of the lambda tests I have written locally so I will add some more code to resolve this later in the week.

**Pull request merged**